esp32-wroom-rp/src/protocol.rs

```diff
@@ -95,7 +102,92 @@ pub(crate) struct NinaLargeArrayParam {
     data: Vec<u8, MAX_NINA_PARAM_LENGTH>,
 }
 
-impl NinaParam for NinaByteParam {
+pub(crate) struct NinaAbstractParam {
+    // Byte representation of length of data
+    length_as_bytes: [u8; 2],
+    // Data to be transfered over SPI bus
+    data: Vec<u8, MAX_NINA_PARAM_LENGTH>,
+    // Number of bytes in data
+    length: u16,
+    // The number of bytes needed to represent
+    // length_as_bytes
+    length_size: u8,
+}
+
+impl NinaParam for NinaAbstractParam {
+    fn length_as_bytes(&self) -> [u8; 2] {
+        self.length_as_bytes
+    }
+
+    fn data(&self) -> &[u8] {
+        self.data.as_slice()
+    }
+
+    fn length(&self) -> u16 {
+        self.length as u16
+    }
+
+    fn length_size(&self) -> u8 {
+        self.length_size
+    }
+}
+
+impl From<NinaNoParams> for NinaAbstractParam {
```

Comment:
@calebbourg I love this so much, it looks and feels so much better to have each sub-type be nothing more than `From` trait implementations.